src/syft/core/message/syft_message.py

```diff
@@ -1,7 +1,69 @@
 from ...common.message import AbstractMessage
+from ...common.id import UID
+
+
+class CommunicationEnvelop(object):
+    """
+    A syft message contains information about who should the message
+    be communicated with next. This information includes:
+    - As a worker receiving this message, should I forward this message?
+        if so to who (specific routes or broadcast)
+        and how (on what protocols)?
+    - As a worker receiving this msg, should I sign the message before forwarding?
+           eg. if the user is keeping count on the number of forwards.
+    - As a worker receiving this message, If this is a processing message,
+        and I need to inform some party when I'm done processing,
+        who should I inform and how?
+    - As a worker sending this message, I should attach this object
+        to my message.
+
+    All the above assumes that worker of this message has visiblity to the workers
+    its been asked to communicate with. If it has no visibility it'll log a warning
+    and
+    """
+    def __init__(self, should_forward: Boolean = False):
+        self.should_forward = should_forward
+        self.should_publish_outcomes = should_publish_outcomes
+        self.forward_routes = []
+        self.forward_broadcast = None
+        self.outcomes_routes = []
+        self.outcomes_broadcast = []
+
+    def is_valid(self):
+        """
+        Ensure the following is met:
+          - if should_forward is True > forward routes or broadcast should be
+            defined.
+          - if should_publish_outcomes is True > outcomes routes or broadcast
+            should be defined.
+        """
+        return True
 
 
 class SyftMessage(AbstractMessage):
-    def __init__(self, route, msg_id=None):
+    def __init__(self, route: Route, comms: CommunicationEnvelop, msg_id: UID = None) -> None:
```

Comment:
What do you see as the primary difference between Route and CommunicationEnvelop?

Reply:
From my perspective, Route should contain the information about where a message should be sent (all in one place). It seems CommunicationEnvelop is also trying to do this, yes?

Reply:
The way I think of a route is it's the path to a single worker, might be potentially overloaded with a communication protocol on how to communicate with this worker - but the worker may be able to communicate on multiple protocols, so all in this case would be defined. In the real world, this could be the external and internal IP to the VM processing a message or publishing a message.

A communication envelop, is in nature very specific to a single message floating around between workers, and its function is to tell workers what to do with this message once they're done processing it, such as send me back the outcomes on http or can you forward this message to 10 other workers to calculate the average age of people you have in your dataset, then once done worker X is going to be your aggregator, or we want one of you to volunteer as an aggregator and send me the message back. All that stuff is facilitated by an envelop because an envelop tells this group of workers how to communicate (protocol) and what to communicate (outcomes, this msg, etc.)